### ISSUE TITLE
"Function result is unused" false-positive for methods returning $this

### ISSUE THREAD

**Issue by someniatko** [kalessil/phpinspectionsea#1087]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Issue type     | False-positive                         |
| Plugin         | Php Inspections (EA Ultimate) 2.0.9 |
| Language level |  PHP 7.2  |

Some methods use so-called fluent interface, where you can call methods of single object in chain like this:
```php
$entity->setName('Name')->setId(1);
```
It is accomplished by returning `$this`. However it is not intended that you have to always use the returned $this further, but the inspection "Function result is unused" still complaints. Probably the inspection should not be fired in case of methods returning $this (it could be probably check if the method has `@return $this` PHPDoc typehint, or has `:self` or `:SameClassName` return typehint.

Current behaviour (description/screenshot:)
--
![symfony-command](https://user-images.githubusercontent.com/15856706/47840302-91890180-ddbe-11e8-9a28-c97766dd0edf.PNG)

Expected behaviour
--
No inspection is fired in case of method returning $this.

Environment details
---
PhpStorm 2018.2.5
Build #PS-182.4892.16, built on October 11, 2018
Licensed to <me>
Subscription is active until <date>
JRE: 1.8.0_152-release-1248-b8 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Windows 10 10.0

**Comment by someniatko** [comment]
#1083 was a duplicate, i guess?

**Comment by kalessil** [comment]
Yep, duplicates #1083.
